test/gtest/ucp/test_ucp_ep_reconfig.cc

```diff
@@ -375,67 +422,43 @@ void test_ucp_ep_reconfig::run(bool bidirectional)
     auto r_sender   = static_cast<const entity*>(&sender());
     auto r_receiver = static_cast<const entity*>(&receiver());
 
-    EXPECT_NE(r_sender->is_reconfigured(), r_receiver->is_reconfigured());
+    if (should_reconfigure()) {
+        EXPECT_NE(r_sender->is_reconfigured(), r_receiver->is_reconfigured());

```

Comment:
maybe split it up to 2 checks (like in `else` branch) to clarify expectation which side should be reconfigured?